rero_ils/modules/operation_logs/api.py

```diff
@@ -17,76 +17,131 @@
+            refresh to make this operation visible to search, if `false`
+            (the default) then do nothing with refreshes.
+            Valid choices: true, false, wait_for
+        :returns: A new :class:`Record` instance.
         """
-        search = OperationLogsSearch()
-        search = search.filter('term', record__pid=record_pid)\
-            .filter('term', record__type=pid_type)\
-            .filter('term', operation=OperationLogOperation.CREATE)
-        oplgs = search.source(['pid']).scan()
-        try:
-            return OperationLog.get_record_by_pid(next(oplgs).pid)
-        except StopIteration:
-            return None
+        record = cls(
+                data,
+                model=None,
+                **kwargs
+            )
+
+        # TODO: enable this whe a invenio-records is updated.
+        # # Run pre create extensions
+        # for e in cls._extensions:
+        #     e.pre_create(record)
+
+        res = current_search_client.index(
+            cls.get_index(record),
+            record.dumps(),
+            refresh=index_refresh)
+        record['id'] = res.get('_id')
+
+        # Run post create extensions
+        for e in cls._extensions:
+            e.post_create(record)
+        return record
 
+    @classmethod
+    def get_index(cls, data):
+        """Get the index name given the data.
+
+        One index per year is created based on the data date field.
+
+        :param data: Dict with the record metadata.
+        :returns: str, the corresponding index name.
+        """
+        suffix = '-'.join(data.get('date', '').split('-')[0:1])
+        return f'{cls.index_name}-{suffix}'
 
-class OperationLogsIndexer(IlsRecordsIndexer):
-    """Operation log indexing class."""
+    @classmethod
+    def bulk(cls, data):
+        """Bulk indexing.
 
-    record_cls = OperationLog
+        :params data: list of dicts with the record metadata.
+        """
+        actions = []
+        for d in data:
+            d = OperationLog(d)
+            action = {
+                '_op_type': 'index',
+                '_index': cls.get_index(d),
+                '_source': d.dumps()
+            }
+            actions.append(action)
+        res = bulk(current_search_client, actions)
+        # TODO: check errors
+        print(res)
 
-    def bulk_index(self, record_id_iterator):
-        """Bulk index records.
+    @classmethod
+    def get_record(cls, id_):
+        """Retrieve the record by ID.
 
-        :param record_id_iterator: Iterator yielding record UUIDs.
+        Raise a database exception if the record does not exist.
+        :param id_: record ID.
+        :returns: The :class:`Record` instance.
         """
-        super(OperationLogsIndexer, self).bulk_index(
-            record_id_iterator, doc_type='oplg')
+        source = current_search_client.get(id=id_, index=cls.index_name)
+        return cls(source.get('_source', {}))
+
+    @classmethod
+    def get_indices(cls):
```

Comment:
Something like that can be used https://www.elastic.co/guide/en/elasticsearch/reference/7.12/cat-indices.html instead of a search ?